pages/Surveypage.js

```diff
@@ -530,8 +639,268 @@ export class Surveypage {
+  await this.page.getByRole('button', { name: 'Q7 Group Question Group' }).click();
+  await expect(this.page.getByPlaceholder('Enter question', { exact: true })).toHaveValue('Group Question');
+  await expect(this.page.locator('.MuiStack-root > div > div > div:nth-child(2) > div > div:nth-child(2) > div > div').first()).toBeVisible();
+  await expect(this.page.getByRole('button', { name: 'Q8 Single Select Visible Tell' })).toBeVisible();
+  await expect(this.page.getByRole('button', { name: 'Q9 Multi Select Visible What' })).toBeVisible();
+}
+async validateErrorsInQuestionnairePage(){
+  await this.page.click(this.addNewQuestionButton);
+  await this.page.getByRole('button', { name: 'Single Select' }).click();
+  await this.page.getByPlaceholder('Enter question', { exact: true }).fill('What are you doing');
+  await this.page.getByRole('button', { name: 'Save' }).click();
+  await expect(this.page.locator('#root')).toContainText('Please ensure that all the required fields are filled in before saving the question.');
+  await this.page.getByPlaceholder('Enter code').fill('1');
+  await this.page.getByRole('button', { name: 'Save' }).click();
+  await expect(this.page.locator('#root')).toContainText('Please ensure that the question code is unique.');
+  await this.page.locator('.MuiInputBase-root').first().click();
+  await this.page.getByPlaceholder('Enter question code').fill('Q100');
+  await this.page.locator('button[name="addButton"]').click();
+  await this.page.getByPlaceholder('Enter code').nth(1).fill('1');
+  await this.page.getByRole('button', { name: 'Save' }).click();
+  await expect(this.page.locator('#root')).toContainText('Please ensure that the option code is unique.');
+  await expect(this.page.getByText('Enter unique code.').first()).toBeVisible();
+  await expect(this.page.locator('#root')).toContainText('Enter unique code.');
+}
+
+async validateErrorsInQuestionnairePageArchiveProject(){
+  await this.page.click(this.addNewQuestionButton);
+  await this.page.getByRole('button', { name: 'Single Select' }).click();
+  await this.page.getByPlaceholder('Enter code').fill('1');
+  await this.page.getByRole('button', { name: 'Save' }).click();
+  await expect(this.page.getByRole('heading')).toContainText('Failed to Delete Question');
+  await expect(this.page.getByLabel('Failed to Delete Question')).toContainText('The project has been archived, no further modifications can be made.');
+  await expect(this.page.getByRole('img')).toBeVisible();
+  await expect(this.page.getByRole('button', { name: 'Go to Home' })).toBeVisible();
+  }
+
+
+async uploadRules(csv_filepath) {
     await this.page.getByRole("button", { name: "Upload Rules" }).click();
     await expect(this.page.getByRole("heading")).toContainText("Upload Rules");
     await expect(this.page.getByLabel("Upload Rules")).toContainText(
```

Comment:
There are still many tests missing here. For example what happens when we refresh, also while I am adding a question, what happens if I click on Add New Question button or click on some other question

Make a list of all the tests missing and share it with me.